tracim/tracim
# Easy test

See #4020, closed after review in favor of #4051 to rename the branch.

Simplify test (and dev backend) launch by:
- moving all needed services into a docker compose file
- allow choosing the database used in backend tests by a `pytest` argument (`--database`)
- adding a script to launch a dev backend server and optionally the cypress tests (`run_dev_backend.sh`).

## Checkpoints



**For developers**

  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**


**For testers**


**Testing method**

- check that running
```shell
cd backend
docker-compose up -d
pytest
```
works (you can try with `pytest --database=postgresql` too)
- check that running
```shell
./run_dev_backend.sh
```
starts a functional backend on http://localhost:7999
- check that running
```shell
./run_dev_backend.sh cypress open
./run_dev_backend.sh cypress run
```
respectively opens cypress and runs the cypress tests in the console.
